wallets/core/src/wallet.ts

```diff
@@ -109,10 +109,18 @@ class Wallet<InstanceType = any> {
           // @ts-ignore
           network: requestedNetwork,
           newInstance: this.tryGetInstance.bind(this),
+          getState: this.getState.bind(this),
         });
 
-        // We assume if we reach here (`switchNetwork` not throwing error), Switch successfully has been done.
-        if (requestedNetwork !== this.state.network) {
+        /*
+         * We assume if we reach here (`switchNetwork` not throwing error), Switch successfully has been done.
+         * But for providers with async switch network like wallet-connect, we need to wait for chain change
+         * event before changing network.
+         */
+        if (
+          requestedNetwork !== this.state.network &&
+          !this.options.config.isAsyncSwitchNetwork
```

Comment:
To make it more general I suggest change it to something like this:
```
this.options.config.avoidUpdateOnSwitchNetwork({
  instance: this.provider,
  network: requestedNetwork,
  getState: this.getState.bind(this),
})
```

Reply:
I will fix this comment and next related ones in my next PR.